BLSQ/iaso
# IA-2183: Refactor org unit validation status

initial demand was to add 'CLOSED' status to the org unit validation status list.
When you go deeper, this list was hard-coded a bit everywhere in the front-end

⚠️ UPDATE: CLOSED status has been removed, but the refactor can be merged

Related JIRA tickets : IA-2183,  IA-2200
## Self proofreading checklist


## Changes

- create api to get list of validation status keys from the api
- hook to get this list, prevent crashing if a status is not translated
- remove all hard coded version of this list and plug it to the hook
- change validation status for links page to fit with the api Links (validated or not)


## How to test

- Visit org unit list, details page and play with validation status, check that 'CLOSED' is an option and is translated
- go to links page (likns list and link tab on org unit detail page) and play with validation status too

## Print screen / video


https://github.com/BLSQ/iaso/assets/12494624/30f78995-6811-4a37-8a42-d15ffbb92cfd
